apps/ergw_core/src/ergw_core_config.erl

```diff
@@ -56,7 +70,26 @@ get(_, _, _) ->
     undefined.
 
 put(Key, Val) ->
-    ok = application:set_env(ergw_core, Key, Val).
+    case is_global_key(Key) of
+	true ->
+	    {ok, _} = ergw_global:put(Key, Val),
```

Comment:
The `ergw_global:put/2` can return also `timeout` - maybe make sense handle it?

Reply:
in that case the system should fail anyway. Remember, in Erlang you code the ok case and let it crash on errors.